backend/tutorings/tests.py

```diff
@@ -0,0 +1,340 @@
+        res = self.retrieve(1)
+        self.assertStatusCode(res, 401)
+
+    def test_if_tutoring_does_not_exist_then_404(self):
+        for user in self.ALL_USERS:
+            self.login(user)
+            res = self.retrieve(42)
+            self.assertStatusCode(res, 404)
+
+    def test_if_logged_in_then_can_retrieve_own_tutoring(self):
+        for user in self.ALL_USERS:
+            self.login(user)
+
+            tutoring_id = Tutoring.objects.filter(applications__state="ACCEPTED")[0].id
+            res = self.retrieve(tutoring_id)
+            self.assertStatusCode(res, 200)
+
+    def test_if_logged_in_then_can_retrieve_published_tutoring(self):
+        for user in self.ALL_USERS:
+            self.login(user)
+
+            for tutoring in Tutoring.objects.filter(
+                    state="ACCEPTED"
+            ):
+                res = self.retrieve(tutoring.id)
+                self.assertStatusCode(res, 200)
+
+    def test_if_admin_then_can_retrieve_any_tutoring(self):
+        self.login("17admin")
+
+        for tutoring in Tutoring.objects.all():
+            res = self.retrieve(tutoring.id)
+            self.assertStatusCode(res, 200)
+
+    def test_if_not_admin_then_cannot_retrieve_forbidden_tutorings(self):
+        self.login("17simple")
+
+        for tutoring in Tutoring.objects.exclude(state="ACCEPTED"):
+            res = self.retrieve(tutoring.id)
+            self.assertStatusCodeIn(res, [403, 404])
+
+    def test_if_admin_then_can_retrieve_all_fields(self):
+        self.login("17admin")
+
+        for tutoring in Tutoring.objects.all():
+            res = self.retrieve(tutoring.id)
+            self.assertStatusCode(res, 200)
+            self.assertSetEqual(
+                set(res.data.keys()),
+                {
+                    "id",
+                    "name",
+                    "contact",
+                    "state",
+                    "publication_date",
+                    "admin_comment",
+                    "place",
+                    "subject",
+                    "level",
+                    "frequency",
+                    "time_availability",
+                    "description",
+                    "applications",
+                },
+            )
+
+    ##########
+    # CREATE #
+    ##########
+
+    create_data = {
+        "name": "Pierre",
+        "contact": "nom.prenom@example.com",
+        "place": "15 eme arrondissement",
+        "subject": "Maths",
+        "level": "Terminale S",
+        "frequency": "1 fois par semaine",
+        "time_availability": "Jeudi soir 18h",
+        "description": "Travailler pour le bac",
+    }
+
+    create_data_with_extra_fields = dict(create_data)
+    create_data_with_extra_fields.update(
+        {
+            "state": "REVIEWING",
+            "publication_date": date(2019, 1, 1),
+            "admin_comment": "ok",
+        }
+    )
+
+    def check_last_tutoring(self, data):
+        tutoring = Tutoring.objects.last()
```

Comment:
Order the objects before calling last. With pagination it may change because we generally use reverse order by default 
ˋorder_by(« id »)`